README.md

```diff
@@ -308,3 +309,112 @@ compose(
   cacheFetch(slave, strategy2, cache2)
 )
 ```
+
+# Graph
+
+## make
+
+signature: `make(<graph description>: GraphDescription): Graph`
+
+*TODO*
+
+## query
+
+signature: `query(graph: Graph, Ps: List[string], A: Dict[string,any]): Observable<>`
+
+`query` accepts the `P`s to query and the `A`s to use as arguments:
+
+- `graph` is the entire graph description obtained with `make()`
+
+- `Ps` is an array of strings corresponding to compound node ids, e.g. `['A', 'C']`
+
+- `A` is an object in the form `{ a1: v1, ... }`, and it should contain all the possible `A`s any fetch we are requesting could need to run
+
+*TODO*
+
+## invalidate
+
+signature: `invalidate(graph: Graph, invalidatePs: List[string], A: Dict[string,any])`
+
+`invalidate` accepts all the `P`s that should be invalidated, for the given `A`s:
+
+- `graph` is the entire graph description obtained with `make()`
+
+- `invalidatePs` is an array of strings corresponding to compound node ids, e.g. `['A', 'C']`
+
+- `A` is an object in the form `{ a1: v1, ... }`, and it should contain all the possible `A`s any fetch we are invalidating could need to run
+
+When invoking `invalidate`, we should provide explicitly everything that needs to be force-refetched. There's no automatic deletion of *dependencies* when invalidating a node, while there's automatic invalidation of all *dependents*:
+
+  - invalidation of all *dependents* will be automatic (all of the *downset* of any invalidated node is recursively invalidated)
```

Comment:
As you use the term `downset` a lot maybe it would be useful to explicitly define it somewhere